rss2email/feed.py

```diff
@@ -910,3 +911,21 @@ def _send_digest(self, digest, seen, sender, send=True):
             if guid not in self.seen:
                 self.seen[guid] = {}
             self.seen[guid]['id'] = id_
+
+    def _fix_user_agent(self):
+        """Fix feed user agent settings broken in 3.11
+
+        Release 3.11 added code to allow the user agent string to be
+        configured globally and on a per-feed basis. It was supposed
+        to allow __VERSION__ and __URL__ in configuration setting to
+        be substituted dynamically with the current version and URL of
+        rss2email, but a bug in the implementation caused them to be
+        substituted statically in the settings for any newly added
+        feed. We've fixed that problem, but we want to go back now and
+        repair feeds that got the wrong user agent value in the
+        interim.
+        """
```

Comment:
Am I reading this right, and instead of `__VERSION__` the previous code would serialize the substituted value into the configuration?

Reply:
Yes. Here's what happens in the v3.11 code (without this PR) when a feed is added:

1. Config is loaded.
2. `__VERSION__` and `__URL__` are substituted into the user agent string at that time.
3. When the config for the feed is saved, the code recognizes that the value of this setting for the feed -- since it has been substituted -- is different for the value in the `DEFAULT` section, so it saves the feed-specific value.